lib/src/main/scala/com/worksap/nlp/uzushio/lib/filters/NoContentDOM.scala

```diff
@@ -1,26 +1,80 @@
 package com.worksap.nlp.uzushio.lib.filters
 
-import com.worksap.nlp.uzushio.lib.cleaning.Paragraph
+import com.worksap.nlp.uzushio.lib.cleaning.{Paragraph, PathSegment}
 import com.worksap.nlp.uzushio.lib.filters.base.ParagraphFilter
 
 class NoContentDOM extends ParagraphFilter {
-  // This names are tag names, but also class names and id names
   final private val filteringDomNames: Seq[String] =
-    Array("header", "footer", "aside", "nav", "left-box", "sidebar")
+    Array("header", "footer", "aside", "nav", "noscript", "form")
+
+  final private val DOMCandidatesForFiliteringClassOrId = Array("div", "p", "ul", "h1")
+
+  final private val filteringFullMatchClassOrIdCandidates: Seq[String] = Array(
+    "left-box",
+    "blog-title-inner",
+    "blogtitle",
+    "blog-name",
+    "head-block1",
+    "head-blog-name",
+    "head-introduction",
+  )
+
+  final private val filteringPartialMatchClassOrIdNames: Seq[String] = Array(
+    "header",
+    "footer",
+    "side",
+    "menu",
+    "nav",
+    "banner",
+    "logo",
+    "pankuzu",
+    "breadcrumb",
+    "widget",
+    "button",
+  )
+
+  final private val filteringFullMatchClassOrIdNames =
+    filteringPartialMatchClassOrIdNames ++ filteringFullMatchClassOrIdCandidates ++ filteringFullMatchClassOrIdCandidates
+      .map(toCamelCase)
+
+  def toCamelCase(s: String): String = {
+    val words = s.split("[_-]")
+    words.head + words.tail.map(_.capitalize).mkString
+  }
+
+  def partialMatchIds(css: PathSegment): Boolean = {
+    if (css.id == null) {
+      return false
+    }
+
+    filteringPartialMatchClassOrIdNames.exists(name => css.lowerId.contains(name))
+  }
+
+  def partialMatchClasses(css: PathSegment): Boolean = {
+    filteringPartialMatchClassOrIdNames.exists(name => css.lowerClasses.exists(_.contains(name)))
```

Comment:
Although these codes may be computationally expensive, I can't catch a good idea.
Please, give me a good solution if you have one.